addons/web_tour/static/src/js/tip.js

```diff
@@ -113,16 +124,100 @@ var Tip = Widget.extend({
         if (!$anchor.is(this.$anchor)) {
             this._unbind_anchor_events();
             this.$anchor = $anchor;
-            this.$ideal_location = this._get_ideal_location();
             if (this.$el.parent()[0] !== document.body) {
-                this.$el.appendTo(this.$ideal_location);
+                if (this.viewPortState === 'in') {
+                    this.$actualAnchor = $anchor;
+                    this.$ideal_location = this._get_ideal_location($anchor);
+                    this.$furtherIdealLocation = this._get_ideal_location(this.$ideal_location);
```

Comment:
The 3 first lines of `attach_to` are equal to those ones => code needs to be shared.